soh/soh/Enhancements/randomizer/hook_handlers.cpp

```diff
@@ -68,7 +69,14 @@ RandomizerCheck GetRandomizerCheckFromSceneFlag(int16_t sceneNum, int16_t flagTy
             (flagType == FLAG_SCENE_COLLECTIBLE && loc.GetCollectionCheck().type == SPOILER_CHK_COLLECTABLE) ||
             (flagType == FLAG_GS_TOKEN && loc.GetCollectionCheck().type == SPOILER_CHK_GOLD_SKULLTULA)
         )) {
-            return loc.GetRandomizerCheck();
+            if (loc.GetQuest() == RCQUEST_BOTH) {
+                return loc.GetRandomizerCheck();
+            } else {
+                auto dungeon = OTRGlobals::Instance->gRandoContext->GetDungeons()->GetDungeonFromScene(loc.GetScene());
+                if (dungeon->IsMQ() == (loc.GetQuest() == RCQUEST_MQ)) {
```

Comment:
Interesting way of matching the two MQ checks here. Pretty sure it works fine, though I'm not sure how others would feel about the clarity of the code. Non-blocking for me, though. Funny that you had to add this, I could have sworn there was a check like this in the previous framework iteration XD.

Reply:
I agree that this also threw me off for a second, but yeah it basically is handling both the "vanilla only" check and the "mq only" check together, thanks to this being in the `else` block after the "both" check.

Reply:
I was hoping that `Dungeon` would have a method to return the quest, ideally in a form compatible with `Location::GetQuest()` so that I could compare them directly. I didn't find any and didn't want to duplicate the check between MQ and Vanilla, so I ended up with that.

I'm more familiar with Rust than C++, and my first instinct here would be to use an exhaustive `match` statement over `loc.GetQuest()`.

Reply:
That probably is a change that needs to happen eventually, but would be more likely to be put in when we start gearing up for modding support.